exercises/zipper/.meta/src/reference/java/Zipper.java

```diff
@@ -0,0 +1,199 @@
+public class Zipper {
+
+    private int value;
+    BinaryTree tree; // Only the root node will bind to the BinaryTree
+    Zipper up;
+    Zipper left;
+    Zipper right;
+
+    public Zipper(int val) {
```

Comment:
As per our POLICIES document, we should limit access to minimal levels.

In that case, we should change this (and other public methods) to package private.